### DISCUSSION TITLE
File sort order in file browser

### DISCUSSION THREAD

**Discussion by Trakyan** [winder/Universal-G-Code-Sender#2227] [Help] [answered]
I'm using the latest nightly build (2.0-SNAPSHOT / 2023-05-25), tried this on both my usual linux machine (Endavour OS, an arch based distro) and the windows 10 computer we use at work (where UGS is actually going to get used). 

When I go to open a gcode file with the open button on the toolbar, it opens the little file explorer and by default it sorts (i think?) by alphabetical order of the file names. I prefer to have it sort by newest to oldest date modified, since I'm usually looking to cut the gcode file I just made. I know this can be done in the file explorer by clicking ''details' and then clicking the "modified" header to sort by date modified, but this setting doesn't persist after closing and restarting UGSplatform. Is there any way to set the default sort order? 

I'm wondering if anyone else experiences this (I'd expect the sort order remains as what it was last changed to, at least that's what other programs seem to do), or if this is a symptom of the fact I've been deleting and reinstalling various versions of UGS on these computers trying to chase down other issues, and I've been unable to purge all the old settings and somethings gotten corrupted between versions. For instance the macros seem to be stored somewhere other than the ugsplatform file that comes out of the zip file and also not in appdata/roaming on a windows machine since they persist to the next "install" when i delete both those off the windows machine.

Just trying to chase down this one last little gremlin, otherwise UGS is working great now, if it's just the way UGS works then that's no big deal for me really.

Thanks in advance for any help.

**Comment by breiler** [comment, marked as answer]
The file browser is something that the Java language is responsible for, trying to simulate the operating systems dialog as much as possible. So the dialog will look and work differently on Mac/Linux/Windows.

The good thing about this is that we get all of these features for free. The downside is that there isn't a generic programming API for customizing it. So there is no way for us to retrieve or set any of settings you make in the dialog. 

This is more of a Java thing rather than an UGS thing which you will have to learn and live with... =(

**Comment by Trakyan** [comment]
Thanks, honestly it's refreshing to hear I'm not responsible for this one haha.
